setup-di-compile/action.yml

```diff
@@ -0,0 +1,60 @@
+name: "Magento compilation (setup:di:compile)"
+author: "MageOS"
+description: "A Github Action that runs bin/magento setup:di:compile."
+
+inputs:
+  php_version:
+    required: true
+    default: "8.1"
+    description: "PHP version used to run setup:di:compile."
+
+  composer_version:
+    required: true
+    default: "2"
+    description: "The version of composer to use."
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout Project
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
```

Comment:
`fetch-depth` should be `1` we don't need the whole history to `di:compile`

Reply:
In fact, you can omit the whole key, `fetch-depth` is `1` by default.

Reply:
@damienwebdev do you think the complete "checkout" action shouldn't be here?